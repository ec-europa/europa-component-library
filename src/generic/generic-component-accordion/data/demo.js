module.exports = function context(system) {
  const c = {
    _demo: {
      scripts: `document.addEventListener('DOMContentLoaded', function () {
        ECL.accordions();
      });`,
    },
    panels: [
      {
        id: 'pan1',
        isExpanded: true,
        heading: {
          title:
            'Jobs, Growth, Investment and Competitiveness with an additional quite long string',
          icon: {
            icon_path: `../../${system}-preset-website/images/icons/symbol-defs.svg`,
            icon: 'growth',
            size: 'xs',
          },
          level: 2,
        },
        body:
          '<p class="ecl-paragraph">Bullfights. Bull hockey. Do you like this? The bull is stabbed, prodded, beaten. The bull is wounded. The bull is tired before the matador ever steps into the ring. Now, is that victory? Of course it is. Wanna know the secret to winning? Creative sportsmanship. In other words, one has to rig the game.</p>',
      },
      {
        id: 'pan2',
        heading: {
          title: 'Energy Union',
          icon: {
            icon_path: `../../${system}-preset-website/images/icons/symbol-defs.svg`,
            icon: 'energy',
            size: 'xs',
          },
          level: 2,
        },
        body:
          '<p class="ecl-paragraph">Bullfights. Bull hockey. Do you like this? The bull is stabbed, prodded, beaten. The bull is wounded. The bull is tired before the matador ever steps into the ring. Now, is that victory? Of course it is. Wanna know the secret to winning? Creative sportsmanship. In other words, one has to rig the game.</p>',
      },
    ],
  };

  return c;
};
