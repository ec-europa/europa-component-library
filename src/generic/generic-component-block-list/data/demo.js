module.exports = {
  blocks: [
    {
      href: '../../example.html#1',
      title: 'Home',
      description: 'x',
    },
    {
      href: '../../example.html#2',
      title: 'About Us',
      description: 'a',
    },
    {
      href: '../../example.html#3',
      title: 'More Information',
      description: 'gi',
    },
    {
      href: '../../example.html#4',
      title: 'Contact Us',
      description: 'gia',
    },
  ],
};
