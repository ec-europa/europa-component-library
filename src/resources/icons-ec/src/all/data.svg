<svg width="48" height="48" viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M24 18C13.4442 18 8 13.7941 8 12C8 10.2059 13.4442 6 24 6C34.5567 6 40 10.2059 40 12C40 13.7941 34.5567 18 24 18ZM40 24C40 26.1252 34.5567 30 24 30C13.4442 30 8 26.1252 8 24V18C11.7315 20.2539 17.5103 21.7091 24 22C30.4916 21.7091 36.2685 20.2539 40 18V24ZM40 36C40 38.1255 34.5567 42 24 42C13.4442 42 8 38.1255 8 36V30C11.7315 32.2538 17.5103 33.7089 24 34C30.4916 33.7089 36.2685 32.2538 40 30V36ZM24 2C12.9713 2 4 6.31954 4 12V36C4 41.6805 12.9713 46 24 46C35.0287 46 44 41.6805 44 36V12C44 6.31954 35.0287 2 24 2Z" fill="black"/>
<mask id="mask0_1_1085491" style="mask-type:alpha" maskUnits="userSpaceOnUse" x="4" y="2" width="40" height="44">
<path fill-rule="evenodd" clip-rule="evenodd" d="M24 18C13.4442 18 8 13.7941 8 12C8 10.2059 13.4442 6 24 6C34.5567 6 40 10.2059 40 12C40 13.7941 34.5567 18 24 18ZM40 24C40 26.1252 34.5567 30 24 30C13.4442 30 8 26.1252 8 24V18C11.7315 20.2539 17.5103 21.7091 24 22C30.4916 21.7091 36.2685 20.2539 40 18V24ZM40 36C40 38.1255 34.5567 42 24 42C13.4442 42 8 38.1255 8 36V30C11.7315 32.2538 17.5103 33.7089 24 34C30.4916 33.7089 36.2685 32.2538 40 30V36ZM24 2C12.9713 2 4 6.31954 4 12V36C4 41.6805 12.9713 46 24 46C35.0287 46 44 41.6805 44 36V12C44 6.31954 35.0287 2 24 2Z" fill="white"/>
</mask>
<g mask="url(#mask0_1_1085491)">
</g>
</svg>
