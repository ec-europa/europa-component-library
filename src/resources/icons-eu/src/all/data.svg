<?xml version="1.0" encoding="UTF-8" ?>
<svg
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
  width="24pt"
  height="24pt"
  viewBox="0 0 24 24"
  version="1.1"
>
<g id="surface1">
<path
      d="M 12 9.25 C 7.160156 9.25 4.667969 7.324219 4.667969 6.5 C 4.667969 5.675781 7.160156 3.75 12 3.75 C 16.839844 3.75 19.332031 5.675781 19.332031 6.5 C 19.332031 7.324219 16.839844 9.25 12 9.25 Z M 19.332031 12 C 19.332031 12.972656 16.839844 14.75 12 14.75 C 7.160156 14.75 4.667969 12.972656 4.667969 12 L 4.667969 9.25 C 6.375 10.28125 9.027344 10.949219 12 11.082031 C 14.976562 10.949219 17.625 10.28125 19.332031 9.25 Z M 19.332031 17.5 C 19.332031 18.472656 16.839844 20.25 12 20.25 C 7.160156 20.25 4.667969 18.472656 4.667969 17.5 L 4.667969 14.75 C 6.375 15.78125 9.027344 16.449219 12 16.582031 C 14.976562 16.449219 17.625 15.78125 19.332031 14.75 Z M 12 1.917969 C 6.945312 1.917969 2.832031 3.894531 2.832031 6.5 L 2.832031 17.5 C 2.832031 20.105469 6.945312 22.082031 12 22.082031 C 17.054688 22.082031 21.167969 20.105469 21.167969 17.5 L 21.167969 6.5 C 21.167969 3.894531 17.054688 1.917969 12 1.917969 Z M 12 1.917969 "
    />
</g>
</svg>
