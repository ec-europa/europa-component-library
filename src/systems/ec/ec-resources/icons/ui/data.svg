<!-- Generated by IcoMoon.io -->
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="28" height="32" viewBox="0 0 28 32">
<title>data</title>
<path d="M14.075 0.376c-7.791 0-14.075 3.078-14.075 7.022v18.115c0 3.848 6.284 7.022 14.075 7.022s14.043-3.11 14.043-6.958v-18.18c0-3.944-6.38-7.022-14.043-7.022zM25.137 25.513c0 1.218-3.751 4.008-11.062 4.008-7.214 0-11.062-2.789-11.062-4.008v-4.713c2.533 1.603 6.541 2.725 11.062 2.725 4.393 0 8.432-1.026 11.062-2.725v4.713zM25.137 16.503c0 1.218-3.751 4.040-11.062 4.040-7.214 0-11.062-2.822-11.062-4.040v-4.681c2.533 1.571 6.541 2.693 11.062 2.693 4.393 0 8.432-1.026 11.062-2.693v4.681zM14.075 11.437c-7.214 0-11.062-2.822-11.062-4.040s3.751-4.008 11.062-4.008c7.214 0 11.062 2.789 11.062 4.008s-3.848 4.040-11.062 4.040z"></path>
</svg>
