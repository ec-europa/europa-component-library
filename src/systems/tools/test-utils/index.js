const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
