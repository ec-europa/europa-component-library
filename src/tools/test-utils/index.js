const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
