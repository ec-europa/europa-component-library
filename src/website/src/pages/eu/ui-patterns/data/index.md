---
title: Data
type: group
order: 3
---
