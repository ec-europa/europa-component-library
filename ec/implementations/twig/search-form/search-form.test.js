import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';
import data from './demo/data';

describe('EC - Search Form', () => {
  const template =
    '@ecl-twig/ec-component-search-form/ecl-search-form.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra form elements', () => {
      expect.assertions(1);

      const withExtraFormElements = merge(data, {
        extra_form_elements:
          '<input type="hidden" id="custId" name="custId" value="1">',
      });

      return expect(render(withExtraFormElements)).resolves.toMatchSnapshot();
    });
  });
  describe('with validation enabled and missing input data', () => {
    test('returns the right warning message', () => {
      expect.assertions(1);

      const dataCompliance = {
        ...data,
        button: {
          label: '',
        },
        icon: {
          path: '',
        },
        _compliance_: true,
      };

      return expect(render(dataCompliance)).resolves.toMatchSnapshot();
    });
  });
});
