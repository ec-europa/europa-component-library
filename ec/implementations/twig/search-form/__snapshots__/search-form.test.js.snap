// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Search Form Default renders correctly 1`] = `
<test>
  <form
    class="ecl-search-form"
    role="search"
  >
    <div
      class="ecl-form-group"
    >
      <label
        class="ecl-form-label ecl-search-form__label"
        for="search-input"
      >
        Search
      </label>
      <input
        class="ecl-text-input ecl-text-input--m ecl-search-form__text-input"
        id="search-input"
        type="search"
      />
    </div>
    <button
      aria-label="Search"
      class="ecl-button ecl-button--search ecl-search-form__button"
      type="submit"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Search
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-button__icon ecl-button__icon--after"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--search"
          />
        </svg>
      </span>
    </button>
  </form>
</test>
`;

exports[`EC - Search Form Default renders correctly with extra attributes 1`] = `
<test>
  <form
    class="ecl-search-form"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
    role="search"
  >
    <div
      class="ecl-form-group"
    >
      <label
        class="ecl-form-label ecl-search-form__label"
        for="search-input"
      >
        Search
      </label>
      <input
        class="ecl-text-input ecl-text-input--m ecl-search-form__text-input"
        id="search-input"
        type="search"
      />
    </div>
    <button
      aria-label="Search"
      class="ecl-button ecl-button--search ecl-search-form__button"
      type="submit"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Search
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-button__icon ecl-button__icon--after"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--search"
          />
        </svg>
      </span>
    </button>
  </form>
</test>
`;

exports[`EC - Search Form Default renders correctly with extra class names 1`] = `
<test>
  <form
    class="ecl-search-form custom-class custom-class--test"
    role="search"
  >
    <div
      class="ecl-form-group"
    >
      <label
        class="ecl-form-label ecl-search-form__label"
        for="search-input"
      >
        Search
      </label>
      <input
        class="ecl-text-input ecl-text-input--m ecl-search-form__text-input"
        id="search-input"
        type="search"
      />
    </div>
    <button
      aria-label="Search"
      class="ecl-button ecl-button--search ecl-search-form__button"
      type="submit"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Search
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-button__icon ecl-button__icon--after"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--search"
          />
        </svg>
      </span>
    </button>
  </form>
</test>
`;

exports[`EC - Search Form Default renders correctly with extra form elements 1`] = `
<test>
  <form
    class="ecl-search-form"
    role="search"
  >
    <div
      class="ecl-form-group"
    >
      <label
        class="ecl-form-label ecl-search-form__label"
        for="search-input"
      >
        Search
      </label>
      <input
        class="ecl-text-input ecl-text-input--m ecl-search-form__text-input"
        id="search-input"
        type="search"
      />
    </div>
    <button
      aria-label="Search"
      class="ecl-button ecl-button--search ecl-search-form__button"
      type="submit"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        >
          Search
        </span>
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-button__icon ecl-button__icon--after"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--search"
          />
        </svg>
      </span>
    </button>
    <input
      id="custId"
      name="custId"
      type="hidden"
      value="1"
    />
  </form>
</test>
`;

exports[`EC - Search Form with validation enabled and missing input data returns the right warning message 1`] = `
<test>
  <form
    class="ecl-search-form"
    role="search"
  >
    <div
      class="ecl-form-group"
    >
      <label
        class="ecl-form-label ecl-search-form__label"
        for="search-input"
      >
        Search
      </label>
      <input
        class="ecl-text-input ecl-text-input--m ecl-search-form__text-input"
        id="search-input"
        type="search"
      />
    </div>
    <button
      aria-label="Search"
      class="ecl-button ecl-button--search ecl-search-form__button"
      type="submit"
    >
      <span
        class="ecl-button__container"
      >
        <span
          class="ecl-button__label"
          data-ecl-label="true"
        />
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs ecl-button__icon ecl-button__icon--after"
          data-ecl-icon=""
          focusable="false"
        >
          <use
            xlink:href="#general--search"
          />
        </svg>
        <!-- ecl2-compliance: icon -->
        <div
          class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-red ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
        >
          <p
            class="ecl-u-type-color-grey ecl-u-ml-m ecl-u-type-xs"
          >
            [icon] - {{ icon.name }} and {{ icon.path }} properties need to set and filled in.
          </p>
        </div>
      </span>
    </button>
    <!-- ecl2-compliance: button--search -->
    <div
      class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-red ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
    >
      <p
        class="ecl-u-type-color-grey ecl-u-ml-m ecl-u-type-xs"
      >
        [button--search] - {{ label }} property needs to set and filled in.
      </p>
    </div>
  </form>
  <!-- ecl2-compliance: search-form -->
  <div
    class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-red ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
  >
    <p
      class="ecl-u-type-color-grey ecl-u-ml-m ecl-u-type-xs"
    >
      [search-form] -  {{ button.label }} needs to be set and filled in. {{ button.icon_path }} needs to be set and filled in to render the search button icon.
    </p>
  </div>
</test>
`;
